src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2493,14 +2493,20 @@ private void emitProperty(
      * called "entries" that have the right return type.
      */
     private boolean specialCaseMapEntries(String propName, FunctionType propertyType) {
-      if (propertyType.getMaxArity() != 0
-          || !propName.equals("entries")
-          || !propertyType.getReturnType().isTemplatizedType()) {
+      if (propertyType.getMaxArity() != 0 || !propName.equals("entries")) {
         return false;
       }
-      TemplatizedType ttType = propertyType.getReturnType().toMaybeTemplatizedType();
-      ImmutableList<JSType> ttypes = ttType.getTemplateTypes();
-      if (!isTemplateOf(ttType, "IteratorIterable")
+
+      ImmutableList<JSType> ttypes = null;
+      JSType returnType = propertyType.getReturnType();
+
+      // In incremental mode this type is noResolveType, while in total mode it is TemplatizedType
+      // They both have getTemplateTypes, but neither extends the other in the class heirarchy.
+      if (returnType.isTemplatizedType())
+        ttypes = ((TemplatizedType) returnType).getTemplateTypes();
+      if (returnType.isNoResolvedType()) ttypes = ((NoResolvedType) returnType).getTemplateTypes();
```

Comment:
Maybe "else if" here to make it clearer that it's checking a few props of returnType?

Reply:
done